src/objects/zcl_abapgit_object_odso.clas.abap

```diff
@@ -0,0 +1,381 @@
+          lv_isact TYPE abap_bool.
+
+    lv_dsona = ms_item-obj_name.
+
+    CALL METHOD ('CL_RSD_ODSO')=>('FACTORY')
+      EXPORTING
+        i_odsobject = lv_dsona
+      RECEIVING
+        r_r_odso    = lo_odso.
+
+    CALL METHOD lo_odso->('IS_ACTIVE')
+      RECEIVING
+        r_is_active = lv_isact.
+
+    rv_active = lv_isact.
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~is_locked.
+
+    DATA: lv_object TYPE eqegraarg.
+
+    lv_object =  ms_item-obj_name.
+    OVERLAY lv_object WITH '                                          '.
+    lv_object = lv_object && '*'.
+
+    rv_is_locked = exists_a_lock_entry_for( iv_lock_object = 'RSD_S_PROV'
+                                            iv_argument    = lv_object ).
+
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~jump.
+    zcx_abapgit_exception=>raise( |Jump to ODSO is not yet supported| ).
+  ENDMETHOD.
+
+
+  METHOD zif_abapgit_object~serialize.
+
+    DATA: lv_dsonam      TYPE c LENGTH 30,
+          lr_details     TYPE REF TO data,
+          lr_infoobjects TYPE REF TO data,
+          lr_navigation  TYPE REF TO data,
+          lr_indexes     TYPE REF TO data,
+          lr_index_iobj  TYPE REF TO data,
+          ls_return      TYPE bapiret2.
+
+    FIELD-SYMBOLS:
+      <ls_details>      TYPE any,
+      <lv_tstpnm>       TYPE any,
+      <lv_timestmp>     TYPE any,
+      <lv_conttimestmp> TYPE any,
+      <lv_owner>        TYPE any,
+      <lt_infoobjects>  TYPE STANDARD TABLE,
+      <lt_navigation>   TYPE STANDARD TABLE,
+      <lt_indexes>      TYPE STANDARD TABLE,
+      <lt_index_iobj>   TYPE STANDARD TABLE.
+
+    TRY.
+        CREATE DATA lr_details     TYPE ('BAPI6116').
+        CREATE DATA lr_infoobjects TYPE STANDARD TABLE OF ('BAPI6116IO').
+        CREATE DATA lr_navigation  TYPE STANDARD TABLE OF ('BAPI6116NA').
+        CREATE DATA lr_indexes     TYPE STANDARD TABLE OF ('BAPI6116IN').
+        CREATE DATA lr_index_iobj  TYPE STANDARD TABLE OF ('BAPI6116II').
+      CATCH cx_sy_create_data_error.
+        zcx_abapgit_exception=>raise( |ODSO is not supported on this system| ).
+    ENDTRY.
+
+    ASSIGN lr_details->*     TO <ls_details>.
+    ASSIGN lr_infoobjects->* TO <lt_infoobjects>.
+    ASSIGN lr_navigation->*  TO <lt_navigation>.
+    ASSIGN lr_indexes->*     TO <lt_indexes>.
+    ASSIGN lr_index_iobj->*  TO <lt_index_iobj>.
+
+    lv_dsonam = ms_item-obj_name.
+
+    CALL FUNCTION 'BAPI_ODSO_GETDETAIL'
+      EXPORTING
+        odsobject            = lv_dsonam
+      IMPORTING
+        details              = <ls_details>
+        return               = ls_return
+      TABLES
+        infoobjects          = <lt_infoobjects>
+        navigationattributes = <lt_navigation>
+        indexes              = <lt_indexes>
+        indexesinfoobjects   = <lt_index_iobj>.
+
+    IF ls_return-type = 'E'.
+      zcx_abapgit_exception=>raise( |Error when geting details of ODSO: { ls_return-message }| ).
+    ENDIF.
+*chage to method
```

Comment:
?

Reply:
Sorry for this commit, I should leave this on my disc. This is my note to change ASSIGNMENT, to the method like @larshp done in https://github.com/larshp/abapGit/blob/1a896613b4394258f4785bf5fd42800d0b01028b/src/objects/zcl_abapgit_object_iobj.clas.abap#L299